x/mongo/driver/topology/connection.go

```diff
@@ -608,13 +602,13 @@ func (c *Connection) WriteWireMessage(ctx context.Context, wm []byte) error {
 
 // ReadWireMessage handles reading a wire message from the underlying connection. The dst parameter
 // will be overwritten with the new wire message.
-func (c *Connection) ReadWireMessage(ctx context.Context, dst []byte) ([]byte, error) {
+func (c *Connection) ReadWireMessage(ctx context.Context) ([]byte, error) {
```

Comment:
Is there a benifit in returning a new slice over writing to a "dst" slice through the function? Wouldn't returning a new slice require a new memory allocation?